src/main/java/org/javawebstack/validator/Validator.java

```diff
@@ -352,6 +373,19 @@ public ValidationConfig(Field field, List<ValidationRule> rules) {
                 if (r.size() > 0)
                     addMapRules(f, rules, new String[]{name}, r);
             }
+            for (Class<? extends Annotation> annotation : RULE_ANNOTATIONS) {
+                Annotation a = f.getDeclaredAnnotation(annotation);
+                if (a != null) {
+                    List<ValidationRule> r = new ArrayList<>();
+                    try {
+                        Class<?> validatorClazz = Class.forName(annotation.getName() + "$Validator"); // stupid way of doing this;
+                        Constructor<ValidationRule> constructor = (Constructor<ValidationRule>) validatorClazz.getConstructor(annotation);
```

Comment:
The constructor might be private amd getConstructor also hits constructors in parent classes which might lead to unexpected behavior. Use getDeclaredConstructor instead.